src/App.js

```diff
@@ -35,7 +42,23 @@ class App extends Component {
         {console.log(this.state.movies[0])}
         <NavBar />
         <main>
-          <MovieContainer movieArray={this.state.movies} />
+          <Modal
+            isOpen={this.state.selectedMovie !== null}
+            onRequestClose={this.unselectMovie}
+          >
+            {this.state.selectedMovie !== null ? (
+              <MovieModal movie={this.state.selectedMovie} />
+            ) : (
+              false
+            )}
+            <button onClick={this.unselectMovie}>❌</button>
+          </Modal>
+          <MovieContainer
+            movieArray={this.state.movies}
+            movieClicked={(movie) =>
+              this.setState({ ...this.state, selectedMovie: movie })
```

Comment:
Because `setState` will merge changes in state, you don't need to pass in `...this.state` into the object. Instead, you can pass in only the property of state that has changed: `{ selectedMovie: movie }`.